pkgs/applications/graphics/dia/default.nix

```diff
@@ -1,37 +1,69 @@
-{ stdenv, fetchgit, autoconf, automake, libtool, gtk2, pkgconfig, perlPackages,
-libxml2, gettext, python, libxml2Python, docbook5, docbook_xsl,
-libxslt, intltool, libart_lgpl, withGNOME ? false, libgnomeui,
-gtk-mac-integration-gtk2 }:
+{ stdenv
+, fetchFromGitLab
+, meson
+, ninja
+, pkg-config
+, perlPackages
+, glib
+, gtk2
+, libxml2
+, python2
+, docbook_xml_dtd_45
+, docbook-xsl-nons
+, libxslt
+, gettext
+, python3
+, desktop-file-utils
+, zlib
+, cairo
+, gtk-mac-integration-gtk2
+}:
 
 stdenv.mkDerivation {
   pname = "dia";
-  version = "0.97.3.20170622";
+  version = "unstable-2020-04-07";
 
-  src = fetchgit {
-    url = https://gitlab.gnome.org/GNOME/dia.git;
-    rev = "b86085dfe2b048a2d37d587adf8ceba6fb8bc43c";
-    sha256 = "1fyxfrzdcs6blxhkw3bcgkksaf3byrsj4cbyrqgb4869k3ynap96";
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "GNOME";
+    repo = "dia";
+    rev = "26a1103827e9397fff291eecc061b083b61d8140";
+    sha256 = "I5KdxFHRvVvF/MfTa5jxGAizt2tgZhQ/9f5xysaejYc=";
   };
 
-  buildInputs =
-    [ gtk2 libxml2 gettext python libxml2Python docbook5
-      libxslt docbook_xsl libart_lgpl ]
-      ++ stdenv.lib.optional withGNOME libgnomeui
-      ++ stdenv.lib.optional stdenv.isDarwin gtk-mac-integration-gtk2;
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+    gettext
+    python3 # for install script
+    gtk2 # for gtk-update-icon-cache
+    libxslt # for xsltproc
+    desktop-file-utils
+    docbook_xml_dtd_45
+    docbook-xsl-nons
+  ];
 
-  nativeBuildInputs = [ autoconf automake libtool pkgconfig intltool ]
-    ++ (with perlPackages; [ perl XMLParser ]);
+  buildInputs = [
+    glib
+    gtk2
+    libxml2
+    python2
+    zlib
+    cairo
+  ] ++ stdenv.lib.optionals stdenv.isDarwin [
+    gtk-mac-integration-gtk2
```

Comment:
Doesn't seem like this is used:
```
# grep -ri gtk_mac src/
# grep -ri gtk-mac src/
#
```

Reply:
I would expect it to be somehow magically linked. Is it not listed in `nix-store --query --tree` or `nix why-depends`?

Reply:
```
# nix-store --query --tree /nix/store/qflajk99al4hk1893gz6z2m996x6l930-dia-0.97.3.20170622 | grep mac
+---/nix/store/brvfal9n3wi562r98q1rgicxcjlyh7i1-gtk-mac-integration-2.0.8
|   +---/nix/store/brvfal9n3wi562r98q1rgicxcjlyh7i1-gtk-mac-integration-2.0.8 [...]
# nix-store --query --tree /nix/store/jaq7jss0hmlglpsjb4w7bjgp9g7y5rpx-dia-unstable-2020-04-07 | grep mac
#
```

Reply:
Hmm, I can still grep `GtkosxApplication` in the source code. Looks like `HAVE_MAC_INTEGRATION` was not ported to Meson.

Reply:
Strangely enough, old dia also doesn't have the bug:

<img width="1005" alt="Screenshot 2020-04-07 at 17 01 29" src="https://user-images.githubusercontent.com/245573/78719221-94f8b480-78f1-11ea-9221-ed06863bb4e8.png">

And it does use the system menubar, like it's supposed to do with gtk-mac-integration.

Reply:
Does the latest change help?

Reply:
@jtojnar The build is now failing with
```
../app/interface.c:1060:34: error: use of undeclared identifier 'dia_app_icon'
        gdk_pixbuf_new_from_inline (-1, dia_app_icon, FALSE, NULL));
                                        ^
1 error generated.
```
After applying
```
diff --git a/app/interface.c b/app/interface.c
--- a/app/interface.c
+++ b/app/interface.c
@@ -1057,7 +1057,7 @@ _create_mac_integration (GtkWidget *menubar)
     gtk_widget_hide (menubar); /* not working, it's shown elsewhere */
     /* setup the dock icon */
     gtkosx_application_set_dock_icon_pixbuf (theOsxApp,
-       gdk_pixbuf_new_from_inline (-1, dia_app_icon, FALSE, NULL));
+       pixbuf_from_resource ("/org/gnome/Dia/icons/org.gnome.Dia.png"));
   }
   /* Don't quit without asking to save files first */
   g_signal_connect (theOsxApp, "NSApplicationBlockTermination",
```
I get it to compile and run. I now get a proper dia icon in the dock and system-wide menu works again. There is no improvement with rendering.